pkg/webhook/webhook.go

```diff
@@ -110,28 +115,39 @@ func (w *Webhook) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	logrus.Debugf("Webhook payload %+v", payload)
+
 	if err != nil {
 		logAndReturn(rw, err)
 		return
 	}
 
-	var revision string
+	var revision, branch, tag string
 	var repoURLs []string
 	// credit from https://github.com/argoproj/argo-cd/blob/97003caebcaafe1683e71934eb483a88026a4c33/util/webhook/webhook.go#L84-L87
 	switch t := payload.(type) {
 	case github.PushPayload:
+		branch, tag = getBranchTagFromRef(t.Ref)
 		revision = t.After
 		repoURLs = append(repoURLs, t.Repository.HTMLURL)
 	case gitlab.PushEventPayload:
+		branch, tag = getBranchTagFromRef(t.Ref)
 		revision = t.CheckoutSHA
 		repoURLs = append(repoURLs, t.Project.WebURL)
 	case gitlab.TagEventPayload:
+		branch, tag = getBranchTagFromRef(t.Ref)
 		revision = t.CheckoutSHA
 		repoURLs = append(repoURLs, t.Project.WebURL)
+	// https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push
 	case bitbucket.RepoPushPayload:
 		repoURLs = append(repoURLs, t.Repository.Links.HTML.Href)
 		for _, change := range t.Push.Changes {
 			revision = change.New.Target.Hash
+			if change.New.Type == "branch" {
+				branch = change.New.Name
+			} else if change.New.Type == "tag" {
+				tag = change.New.Name
+			}
```

Comment:
nit: would adding the following be overkill?

```go
} else {
    logrus.Errorf("invalid change type detected or something else here: %s", change.New.Type)
}
```

Reply:
The doc only mentioned it would only be either these two, so just adding `else if` to program safely.